docs/DeveloperGuide.md

```diff
@@ -1,25 +1,277 @@
+  anywhere in the application, while the latter is a catch-all "command" for all erroneous commands a user enters. 
+   
+The `Parser` classes play important roles in execution of specific commands, e.g. `CreateCommand`, because
+ they define methods that check and ensure the conformity of user input to the commands' expected input. Below is a
+  sequence diagram showcasing this interaction, for execution of a `CreateCommand`, e.g. `create
+   microeconomics`:
+  
+![DG-Design CreateCommand Sequence UML](./images-dg/DG-Design-Sequence-Diagram.png?raw=true "CreateCommand UML
+ Sequence Diagram")
+
+### Deck Model
+
+![DG-Design Model UML](./images-dg/DG-Design-Model.png?raw=true "Model UML Class Diagram")
+
+**API**: [seedu/ecardnomics/deck](https://github.com/AY2021S1-CS2113-T14-2/tp/tree/master/src/main/java/seedu/ecardnomics/deck)
+
+The Deck Model component is made up of three parts:
+* `DeckList`
+* `Deck`
+* `FlashCard`
+
+The `FlashCard` component represents a flashcard, storing question
+and answer data. The `Deck` represents a collection of flashcards
+related by a common topic. The `DeckList` represents the collection
+of all the `Deck` objects that the user has.
+
+Only the `Command` components can modify the `DeckList`, `Deck` and
+`FlashCard` components. However, `Ui`, `DeckParser` and `NormalParser`
+are able to read data from the `DeckList`, `Deck` and `FlashCard`
+components.
+
+### Storage
+
+#### Loading the deckList data
+
+![Storage Sequence Diagram](./images-dg/Storage.png?raw=true "load Storage sequence diagram")
+
+**API**: [seedu/ecardnomics/storage](https://github.com/AY2021S1-CS2113-T14-2/tp/blob/master/src/main/java/seedu/ecardnomics/storage)
+
+Storage of this application uses basic `.txt` read and write functions.  
+Upon start of the program, the application checks whether there is a `./data` folder and creates one is there isn't.  
+Then, it reads from the storage file `deckList.txt` line by line to create:
+* new `Deck`
+* new `FlashCard`
+
+and adds them to the current `deckList` passed into the `load` method call.
+
+#### Writing the deckList data
+
+Similarly, for writing the data into `.txt` file, the Storage will loop through all the current `Decks` and their
+current `FlashCards` and write them in a specific format in the text file in the `./data` folder.
+
+## Implementation - Features
+
+### Print to PowerPoint SlideShow
+
+An additional feature targeted at students who wish to use add more style to their flash cards outside of the command 
+line option to allow keep things interesting when they are revising.
+
+The `PowerPointCommand` is parsed by `NormalParser` but the "Print to PowerPoint" command can be called from both Normal 
+and Deck Mode. 
+
+The following diagram shows how the `PowerPointCommand`'s `execute()` calls the `createNewPowerPoint()` method of `PowerPoint`.
+![PPTX Sequence Diagram](images-dg/DG-PPTX-Sequence-Diagram.png)
```

Comment:
Are you missing the return arrows?

![image](https://user-images.githubusercontent.com/60382274/97514645-3ba9bd00-19ca-11eb-8e03-ee2c0a704e6c.png)